File: 240409/왔다 갔던 구역 2/area-been-to-and-from2.java
import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
        // 여기에 코드를 작성해주세요.
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        Integer[] arr = new Integer[2000];
        for(int i=0;i<2000;i++) {
            arr[i] = 0;
        }
        int start = 1000;
        for(int i=0;i<n;i++){
            int x = sc.nextInt();
            char dir = sc.next().charAt(0);
            if(dir =='R') {
                for(int j=start+1;j<start+x;j++) {
                    arr[j]++;
                }
                start += x;
            }
            else {
                for(int j=start-1;j>start-x;j--) {
                    arr[j]++;
                }
                start -= x;
            }
        }
        int count = 0;
        for(int i=0;i<2000;i++) {
            if(arr[i] >=2)
                count++;
        }
        System.out.println(count);
    }
}

File: 240419/왔다 갔던 구역 2/area-been-to-and-from2.kt
import java.util.Scanner

fun main() {
    val sc = Scanner(System.`in`)

    val n = sc.nextInt()
    var curX = 0
    val arr = IntArray(2001)

    repeat(n) {
        val x = sc.nextInt()
        val direction = sc.next()

        if (direction == "L") {
            ((curX - x + 1000)..(curX + 1000 - 1)).forEach {
                arr[it]++
            }
            curX -= x
        } else {
            ((curX + 1000)..(curX + x + 1000 - 1)).forEach {
                arr[it]++
            }
            curX += x
        }
    }

    print(arr.count { it >= 2 })
}